lib/active_shipping/shipping/carriers/usps.rb

```diff
@@ -578,22 +582,25 @@ def parse_tracking_with_fields_response(response)
           success = !(tracking_info.elements['Error'] || summary_no_record?(tracking_summary))
           message = response_message(tracking_info)
           shipment_events = []
-          status, expected_delivery_date, actual_delivery_date = nil
+          status, status_category, expected_delivery_date, actual_delivery_date = nil, nil, nil, nil
 
           if success 
+            previous_event_time = Time.now
             tracking_details.each do |event|
-              ship_event = extract_track_with_fields_event(event)
-              shipment_events << ship_event if !ship_event.nil?
+              ship_event = extract_track_with_fields_event(event, previous_event_time)
+              previous_event_time = ship_event.time
+              shipment_events << ship_event
             end
-            summary_ship_event = extract_track_with_fields_event(tracking_summary)
+            summary_ship_event = extract_track_with_fields_event(tracking_summary, previous_event_time)
             if !summary_ship_event.nil?
               shipment_events << summary_ship_event
+              status = summary_ship_event.status
               actual_delivery_date = summary_ship_event.time if summary_ship_event.delivered?
             end
 
             shipment_events = shipment_events.sort_by(&:time)
-            status = tracking_info.elements['StatusCategory'].get_text.to_s
-            status = status.downcase.gsub("\s", "_").to_sym
+            status_category = tracking_info.elements['StatusCategory'].get_text.to_s
```

Comment:
Was this just a previous bug in the library? What is the difference between the status and status category?